R/guide-axis.R

```diff
@@ -314,9 +351,21 @@ GuideAxis <- ggproto(
 
     measure <- params$measure_text
 
-    length <- elements$ticks_length
-    spacer <- max(unit(0, "pt"), -1 * length)
-    labels <- do.call(unit.c, lapply(grobs$labels, measure))
+    # Ticks
+    range <- range(
+      0, convertUnit(elements$major_length, "cm", valueOnly = TRUE)
+    )
+    if (params$minor.ticks && !inherits(elements$minor, "element_blank")) {
+      range <- range(
+        range, convertUnit(elements$minor_length, "cm", valueOnly = TRUE)
+      )
+    }
+
+    length <- unit(range[2], "cm")
+    spacer <- max(unit(0, "pt"), unit(-1 * diff(range), "cm"))
+
+    # Text
+    labels <- do.call(unit.c, lapply(grobs$label, measure))
```

Comment:
Use `inject()` - and if possible keep the computation (`lapply(...)`) out of the `inject` call to avoid complicated debugging situations

Reply:
Found an even simpler method :)